File: Jump Game II.java
public class Solution {
    public int jump(int[] A) {
        int count = 0;
        int start = 0; int end = 1; //range in this step
        while (end < A.length) {
            int farest = start;
            for (int i = start; i < end; i++) {
                int far = i + A[i];
                if (far > farest)
                    farest = far;
            }
            start = end;
            end = farest+1;
            count++;
        }
        return count;
    }
}

File: Jump Game II.py
class Solution:
	# @param A, a list of integers
	# @return an integer
	def jump(self, A):
		step = 0
		start = 0
		end = 1
		while (end < len(A)):
			farest = start
			for i in range(start, end):
				far = i + A[i]
				if (far > farest):
					farest = far
			start = end
			end = farest + 1
			step = step + 1
		return step